cmd/agent/main.go

```diff
@@ -57,7 +57,14 @@ func main() {
 			rOpts := new(reporter.Options).InitFromViper(v, logger)
 			tchanBuilder := tchannel.NewBuilder().InitFromViper(v, logger)
 			grpcBuilder := grpc.NewConnBuilder().InitFromViper(v)
-			cp, err := app.CreateCollectorProxy(rOpts, tchanBuilder, grpcBuilder, logger, mFactory)
+			builders := map[reporter.Type]app.CollectorProxyBuilder{}
+			builders[reporter.GRPC] = app.GRPCCollectorProxyBuilder(grpcBuilder)
+			builders[tchannel.ReporterType] = app.TCollectorProxyBuilder(tchanBuilder)
```

Comment:
```suggestion
			builders := map[reporter.Type]app.CollectorProxyBuilder{
			    reporter.GRPC: app.GRPCCollectorProxyBuilder(grpcBuilder),
			    tchannel.ReporterType: app.TCollectorProxyBuilder(tchanBuilder),
			}
```
can we do this instead?